src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -3551,11 +3585,22 @@ private void onApplicationDataContOrFin(
                         transferBuilder.more(1);
                     }
 
-                    final AmqpTransferFW transfer = transferBuilder.build();
                     final DirectBuffer fragmentBuffer = messageFragment.buffer();
                     final int fragmentOffset = messageFragment.offset();
                     final int fragmentLimit = messageFragment.limit();
                     final int fragmentSize = fragmentLimit - fragmentOffset;
+                    if (maxMessageSizeExceeded)
```

Comment:
Likely we should avoid calling `onApplicationDataContOrFin` again for same deliveryId in case we get `DATA (init)`, `DATA (cont)`, `DATA (fin)` and the max message size is exceeded in `DATA (init)` or `DATA (cont)`.

This could be done from `onApplicationData`, and rather than tracking a `boolean` for `maxMessageSizeExceeded` we could consider adding `abortedDeliveryId` and assigning it to the current `deliveryId` at the time we send `TRANSFER aborted`.

Then in `onApplicationData` we would have something like:
```
                    if ((flags & FLAG_INIT) == FLAG_INIT)
                    {
                        deliveryId++;
                        onApplicationDataInit(traceId, reserved, authorization, flags, extension, payload);
                    }
                    else if (deliveryId != abortedDeliveryId)
                    {
                        onApplicationDataContOrFin(traceId, reserved, authorization, flags, payload);
                    }
```
which means we wouldn't need to reset `maxMessageSizeExceeded` back to false, as `deliveryId != abortedDeliveryId` will implicitly become true for the next delivery, as needed.